src/client/containers/SpecialOfferPage/SpecialOfferPage.container.js

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect, useState } from 'react';
+import './SpecialOfferPage.styles.css';
+import OfferProducts from '../../components/OfferProductsList/OfferProductsList.component';
+import Pagination from '../../components/Pagination/Pagination.component';
+
+export default function SpecialOfferPage() {
+  const [data, setData] = useState([]);
+  const [currentPageIndex, setCurrentPageIndex] = useState(0);
+  useEffect(() => {
+    fetch('/api/products/on_discount')
+      .then((res) => res.json())
+      .then((json) => setData(json));
+  }, []);
```

Comment:
This is fine, but an improvement that could be done in the future would be to integrate with the `useProducts` hook.

Furthermore, the API design could be improved. I would prefer if `onDiscount` was an argument on the product endpoint, rather than being a separate endpoint. That would make it easier to reuse code and to combine a query for products that are discounted with for example sort order or other arguments.

But since the API endpoint was already done, let's not change that now.